zboxcore/sdk/sdk.go

```diff
@@ -888,6 +893,60 @@ func CreateAllocationForOwner(name string, owner, ownerpublickey string,
 	return
 }
 
+func getAllocationBlobbers(owner, ownerpublickey string,
+	datashards, parityshards int, size, expiry int64,
+	readPrice, writePrice PriceRange, mcct time.Duration) ([]string, error) {
+
+	var allocationRequest = map[string]interface{}{
+		"data_shards":                   datashards,
+		"parity_shards":                 parityshards,
+		"size":                          size,
+		"owner_id":                      owner,
+		"owner_public_key":              ownerpublickey,
+		"expiration_date":               expiry,
+		"read_price_range":              readPrice,
+		"write_price_range":             writePrice,
+		"max_challenge_completion_time": mcct,
+	}
+
+	allocationData, _ := json.Marshal(allocationRequest)
+
+	params := make(map[string]string)
+	params["allocation_data"] = string(allocationData)
+
+	allocBlobber, err := zboxutil.MakeSCRestAPICall(STORAGE_SCADDRESS, "/get_alloc_blobbers", params, nil)
+	if err != nil {
+		return nil, err
+	}
+	var allocBlobberIDs []string
+
+	err = json.Unmarshal(allocBlobber, &allocBlobberIDs)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal blobber IDs")
+	}
+
+	return allocBlobberIDs, nil
+}
+func getFreeAllocationBlobbers(request map[string]interface{}) ([]string, error) {
+	data, _ := json.Marshal(request)
+
+	params := make(map[string]string)
+	params["free_allocation_data"] = string(data)
+
+	allocBlobber, err := zboxutil.MakeSCRestAPICall(STORAGE_SCADDRESS, "/get_free_alloc_blobbers", params, nil)
```

Comment:
@dabasov Why do we need an extra endpoint for free blobbers? We can just make a single endpoint accepting filters like read/write price, so here it will just put read/write == 0 and get same data back.

And filters will be used by clients anyway to get most attractive blobbers by price

Reply:
@moldis then we will need to parse marker data on goSDK side and assemble request, now this logic is done on 0chain side. we can create another issue for this kind of refactoring. In my opinion we should send to gosdk payloads, but not byte strings that are already serialized in the zboxcli tool, it's bad architecture for sure

Reply:
@dabasov I am trying to find the get_free_alloc_blobbers endpoint over repo's, but can't. What I meant is to use `get_alloc_blobbers` instead with `"read_price_range":  0, and "write_price_range":  0.` I suppose it will give the same result as `get_free_alloc_blobbers`

Reply:
It's  changed to`free_alloc_blobbers` and `alloc_blobbers` now. The`free_alloc_blobbers` read the ReadPrice and WritePrice config from global config that stores in blockchain, if we call `alloc_blobbers`, we will need to send another request to get global config from blockchain.